pretrain/openwebtext.py

```diff
@@ -158,30 +158,30 @@ def train(fabric: L.Fabric, state: dict, train_dataloader: DataLoader, val_datal
             optimizer.zero_grad()
             state["step_count"] += 1
 
-        total_lengths += input_ids.size(1)
-        if state["iter_num"] % log_interval == 0:
+        if iter_num % log_interval == 0:
             loss_item = loss.item()  # expensive device-to-host synchronization
             t1 = time.perf_counter()
             throughput.update(
                 time=t1 - total_t0,
-                samples=(state["iter_num"] + 1) * micro_batch_size,
-                lengths=total_lengths,
+                batches=iter_num,
+                samples=iter_num * micro_batch_size,
+                lengths=iter_num * micro_batch_size * model.max_seq_length,
                 flops_per_batch=measured_flops,
             )
-            throughput.compute_and_log(step=state["iter_num"])
```

Comment:
Can we keep the stateful pattern please? 
By defining the state and reading and writing to it, we can consistently keep the script stateful. 
The dict access isn't convenient I know, I'd like to do https://github.com/Lightning-AI/lightning/issues/18515 eventually

Reply:
I'm not changing the stateful pattern here at all, just defining a variable for convenience.

The `iter_num` key is still written and increased by the `enumerate` loop

Reply:
I wanted to show the pattern here that you always access the state directly. If you always access the variable through the state, you don't have to think about writing back to it and makes everything consistent.